nmstatectl/nmstatectl.py

```diff
@@ -90,12 +116,49 @@ def _filter_interfaces(state, patterns):
     return showinterfaces
 
 
-def show(args):
-    state = netinfo.show()
-    if args.only != '*':
-        patterns = [p for p in args.only.split(',')]
+def _filter_state(state, whitelist):
+    if whitelist != '*':
+        patterns = [p for p in whitelist.split(',')]
         state['interfaces'] = _filter_interfaces(state, patterns)
+    return state
+
+
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
+
+    editor = os.environ.get('EDITOR', 'vi')
+    if args.yaml:
+        suffix = '.yaml'
+        txtstate = pretty_yaml(state)
+    else:
+        suffix = '.json'
+        txtstate = pretty_json(state)
+    with tempfile.NamedTemporaryFile(suffix=suffix,
+                                     prefix='nmstate-') as statefile:
+        statefile.write(txtstate)
+        statefile.flush()
+
+        try:
+            subprocess.check_call([editor, statefile.name])
 
+            statefile.seek(0)
```

Comment:
If possible, please allow user to regret the `edit` action by delete all lines in the tempfile. That's what git did.

Reply:
This is now handled.